lib/_tls_wrap.js

```diff
@@ -888,91 +877,7 @@ Server.prototype.setSecureContext = function(options) {
   if (options === null || typeof options !== 'object')
     throw new ERR_INVALID_ARG_TYPE('options', 'Object', options);
 
-  if (options.pfx)
-    this.pfx = options.pfx;
-  else
-    this.pfx = undefined;
-
-  if (options.key)
-    this.key = options.key;
-  else
-    this.key = undefined;
-
-  if (options.passphrase)
-    this.passphrase = options.passphrase;
-  else
-    this.passphrase = undefined;
-
-  if (options.cert)
-    this.cert = options.cert;
-  else
-    this.cert = undefined;
-
-  if (options.clientCertEngine)
-    this.clientCertEngine = options.clientCertEngine;
-  else
-    this.clientCertEngine = undefined;
-
-  if (options.ca)
-    this.ca = options.ca;
-  else
-    this.ca = undefined;
-
-  if (options.minVersion)
-    this.minVersion = options.minVersion;
-  else
-    this.minVersion = undefined;
-
-  if (options.maxVersion)
-    this.maxVersion = options.maxVersion;
-  else
-    this.maxVersion = undefined;
-
-  if (options.secureProtocol)
-    this.secureProtocol = options.secureProtocol;
-  else
-    this.secureProtocol = undefined;
-
-  if (options.crl)
-    this.crl = options.crl;
-  else
-    this.crl = undefined;
-
-  if (options.ciphers)
-    this.ciphers = options.ciphers;
-  else
-    this.ciphers = undefined;
-
-  if (options.ecdhCurve !== undefined)
-    this.ecdhCurve = options.ecdhCurve;
-  else
-    this.ecdhCurve = undefined;
-
-  if (options.dhparam)
-    this.dhparam = options.dhparam;
-  else
-    this.dhparam = undefined;
-
-  if (options.honorCipherOrder !== undefined)
-    this.honorCipherOrder = !!options.honorCipherOrder;
-  else
-    this.honorCipherOrder = true;
-
-  const secureOptions = options.secureOptions || 0;
-
-  if (secureOptions)
-    this.secureOptions = secureOptions;
-  else
-    this.secureOptions = undefined;
-
-  if (options.sessionIdContext) {
-    this.sessionIdContext = options.sessionIdContext;
-  } else {
-    this.sessionIdContext = crypto.createHash('sha1')
-                                  .update(process.argv.join(' '))
-                                  .digest('hex')
-                                  .slice(0, 32);
-  }
+  setOptions.call(this, options);
```

Comment:
`setSecureContext()` shouldn't be calling `setOptions()`.

Reply:
But when I change this to `setOptions(options)` directly, plenty of tests failed. It seems that need to call `this` here.
![screen shot 2018-11-28 at 10 37 34 pm](https://user-images.githubusercontent.com/23313266/49158923-454dc480-f35e-11e8-88ba-741555b50c24.png)

Any advice ?